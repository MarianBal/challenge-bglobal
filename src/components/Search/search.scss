@import '../../resources/mixin.scss';
@import '../../resources/colors.scss';
@import '../../resources/medley.scss';

.search-container {
  width: 70%;
  padding: 15px 0;
  @include flexbox(row, center, center);

  .input-container {
    width: 200px;
    height: 30px;
    border: solid 1px $dark-color;
    border-radius: $border-radius;
    padding: 2px;
    @include flexbox(row, flex-start, center);
  }

  input {
    border: none;
  }

  .fas {
    color: $dark-color;
  }
}
