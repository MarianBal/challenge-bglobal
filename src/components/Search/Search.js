import React, { useState } from 'react';
import { Link } from 'react-router-dom';

import './search.scss';

const Search = () => {
  const [search, setSearch] = useState('');
  const url = `/search/${search}`;

  return (
    <div className='search-container'>
      <div className='input-container'>
        <input
          placeholder='Buscar Pókemon'
          name='search'
          type='text'
          value={search}
          onChange={event => setSearch(event.target.value)}
        />
        <Link to={url}>
          <i className='fas fa-search' alt='Botón de búsqueda' />
        </Link>
      </div>
    </div>
  );
};

export default Search;
