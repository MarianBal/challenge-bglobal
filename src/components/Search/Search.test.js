import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Search from './Search';

configure({ adapter: new Adapter() });

describe('Search render correctly', () => {
  it('Search renders', () => {
    shallow(<Search />);
  });

  it('Search render correctly', () => {
    const wrapper = shallow(<Search />);
    expect(wrapper).toMatchSnapshot();
  });
});
