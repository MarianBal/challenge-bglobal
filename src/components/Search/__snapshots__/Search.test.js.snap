// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search render correctly Search render correctly 1`] = `ShallowWrapper {}`;
